collect_app/src/main/java/org/odk/collect/android/widgets/RatingWidget.java

```diff
@@ -14,121 +14,95 @@
 
 package org.odk.collect.android.widgets;
 
+import android.annotation.SuppressLint;
 import android.app.Activity;
 import android.content.Context;
-import android.graphics.drawable.Drawable;
-import android.util.DisplayMetrics;
-import android.widget.GridLayout;
-import android.widget.ImageButton;
+import android.view.View;
 
 import org.javarosa.core.model.RangeQuestion;
 import org.javarosa.core.model.data.IAnswerData;
 import org.javarosa.core.model.data.IntegerData;
-import org.odk.collect.android.R;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.databinding.RatingWidgetAnswerBinding;
 import org.odk.collect.android.formentry.questions.QuestionDetails;
-import org.odk.collect.android.formentry.questions.WidgetViewUtils;
+import org.odk.collect.android.utilities.ScreenUtils;
+import org.odk.collect.android.utilities.UiUtils;
 
+@SuppressLint("ViewConstructor")
 public class RatingWidget extends QuestionWidget {
 
     public static final int ASSUMED_TOTAL_MARGIN_AROUND_WIDGET = 40;
+    public static final int STANDARD_WIDTH_OF_STAR = 48;
 
-    final GridLayout gridLayout;
-    Integer answer;
+    RatingWidgetAnswerBinding binding;
 
     public RatingWidget(Context context, QuestionDetails questionDetails) {
         super(context, questionDetails);
+    }
 
-        RangeQuestion rangeQuestion = (RangeQuestion) getFormEntryPrompt().getQuestion();
-        int numberOfStars = rangeQuestion.getRangeEnd().intValue();
-        int columns = calculateColumns();
-        int rows = (int) Math.ceil((double) numberOfStars / columns);
-
-        gridLayout = new GridLayout(context);
-        gridLayout.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
-
-        renderGrid(context, numberOfStars, columns, rows);
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        binding = RatingWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+        View answerView = binding.getRoot();
 
-        String answerText = questionDetails.getPrompt().getAnswerText();
-        if (answerText != null) {
-            answer = Integer.parseInt(answerText);
-            for (int i = 0; i < answer; i++) {
-                ImageButton button = (ImageButton) gridLayout.getChildAt(i);
-                button.setImageResource(R.drawable.ic_star);
-            }
-        }
+        RangeQuestion rangeQuestion = (RangeQuestion) prompt.getQuestion();
+        int numberOfStars = rangeQuestion.getRangeEnd().intValue();
```

Comment:
This also could be factored out to a new method (again the calculation, you can keep the variable - as I said below).